components/external_server/tests/utils/test_functions.py

```diff
@@ -32,6 +35,29 @@ def test_md5_hash(self):
         """Test that the hash works."""
         self.assertEqual("bc9189406be84ec297464a514221406d", md5_hash("XXX"))
 
+    def test_sanitize_html(self):
+        """Test that URLs are converted to anchors."""
+        self.assertEqual(f'<p><a href="{self.URL}" target="_blank">{self.URL}</a></p>', sanitize_html(self.URL))
+
+    def test_sanitize_html_does_mot_convert_text_to_html(self):
+        """Test that text without HTML does not get converted to HTML."""
+        self.assertEqual("Text.", sanitize_html("Text."))
+
+    def test_sanitize_html_does_not_change_target(self):
+        """Test that existing target attributes are not changed."""
+        self.assertEqual(
+            f'<a href="{self.URL}" target="_top">{self.URL}</a>',
+            sanitize_html(f'<a href="{self.URL}" target="_top">{self.URL}</a>'),
+        )
+        self.assertEqual(
+            f'<a href="{self.URL}" target="_blank">{self.URL}</a>',
+            sanitize_html(f'<a href="{self.URL}" target="_blank">{self.URL}</a>'),
```

Comment:
Don't you want to test `target="_self"` here instead?

Reply:
Or `target="_top"` as documented

Reply:
It doesn't matter what the value of target is here, the point is that existing target attributes are not changed.